### DISCUSSION TITLE
Attempting to solve / n1 problem for `findMany` (pre-mature optimisation or not?)

### DISCUSSION THREAD

**Discussion by mikevercoelen** [prisma/prisma#16481] [Q&A]
I’m curious how people solve the n1 problem for, for example pagination / findMany queries?

Let’s say I have the following models: User Channel and Event

A channel has events, and each event has a user.

For a paginated findMany query on events, this means Prisma is executing another query PER row? Because I would need to use include: { user }

One “hacky” solution would be to make the user on an event a json field and make it write the user object, when a user updates their info (which doesn’t happen that many times), it would update all that users’s events.

I have read the docs on the n1 query problem, and if I'm correct Prisma uses a similar apollo / graphql style dataloader batcher to batch queries.

Does anyone have any opinions on this? Am I prematurely optimising? Overthinking? Or does this make sense to people?

**Comment by nurul3101** [comment]
Hello @mikevercoelen 👋 

This section in the docs refers to [solving n+1 problem](https://www.prisma.io/docs/guides/performance-and-optimization/query-optimization-performance#solving-the-n1-problem).
Prisma does automatically batch findUnique queries that have the same where and include parameters.

If the data is not going to be updated often, then there is no harm in duplicating it, it would make your reads a lot faster.

**Reply by vladi-stevanovic** [reply]
👋 Hi @mikevercoelen , did you perhaps have a chance to check Nurul's recommendation? Let us know if you still have any questions! 

Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
